pages/docs/tracing-features/masking.mdx

```diff
@@ -275,6 +275,167 @@ def llama_index_fn(question: str):
+from langfuse.decorators import langfuse_context, observe
+from llm_guard.vault import Vault
+from llm_guard.input_scanners import Anonymize
+from llm_guard.input_scanners.anonymize_helpers import BERT_LARGE_NER_CONF
+
+# Step 3: Initialize the Vault and configure the Anonymize scanner
+vault = Vault()
+
+def create_anonymize_scanner():
+    scanner = Anonymize(
+        vault,
+        recognizer_conf=BERT_LARGE_NER_CONF,
+        language="en"
+    )
+    return scanner
+
+# Step 4: Define the masking function
+def masking_function(data):
+    if isinstance(data, str):
+        scanner = create_anonymize_scanner()
+        # Scan and redact the data
+        sanitized_data, is_valid, risk_score = scanner.scan(data)
+        return sanitized_data
+    return data
+
+# Step 5: Configure the masking function
+langfuse_context.configure(mask=masking_function)
+
+# Step 6: Create a sample function with PII
+@observe()
+def generate_report():
+    # Simulated data containing personal names
+    report = "John Doe met with Jane Smith to discuss the project."
+    return report
+
+# Step 7: Observe the trace
+result = generate_report()
+
+print(result)
+# Output: [REDACTED_PERSON] met with [REDACTED_PERSON] to discuss the project.
+```
+
+<Frame border>
+![Redacted trace in Langfuse](/images/docs/masking/masking_example_2.png)
+</Frame>
+
+[Link to the trace in Langfuse 2](https://cloud.langfuse.com/project/cloramnkj0002jz088vzn1ja4/traces/4abb206f-f8fd-4492-86b9-801602513afd?timestamp=2025-01-17T09%3A30%3A04.127Z)
+
+## Example 3: Masking Email and Phone Numbers
+
+You can extend the masking function to redact other types of PII such as email addresses and phone numbers using regular expressions.
+
+```python
+import re
+from langfuse.decorators import langfuse_context, observe
+
+def masking_function(data):
+    if isinstance(data, str):
+        # Mask email addresses
+        data = re.sub(r'\b[\w.-]+?@\w+?\.\w+?\b', '[REDACTED EMAIL]', data)
+        # Mask phone numbers
+        data = re.sub(r'\b\d{3}[-. ]?\d{3}[-. ]?\d{4}\b', '[REDACTED PHONE]', data)
+    return data
+
+langfuse_context.configure(mask=masking_function)
+
+@observe()
+def contact_customer():
+    info = "Please contact John at john.doe@example.com or call 555-123-4567."
+    return info
+
+result = contact_customer()
+
+print(result)
+# Output: Please contact John at [REDACTED EMAIL] or call [REDACTED PHONE].
+```
+
+<Frame border>
+![Redacted trace in Langfuse 3](/images/docs/masking/masking_example_3.png)
+</Frame>
+
+[Link to the trace in Langfuse](https://cloud.langfuse.com/project/cloramnkj0002jz088vzn1ja4/traces/dcc4d640-492e-47a6-b419-922c8b9e0f0f?timestamp=2025-01-17T09%3A38%3A06.814Z)
+
+With custom masking functions in Langfuse, you can effectively control and sanitize the data that gets traced and sent to the server. Whether it's for compliance reasons or to protect user privacy, masking sensitive data is a crucial step in responsible application development.
```

Comment:
Great paragraph, I'd put it at the top as it does not belong to the third example but is a more general statement